nucliadb/nucliadb/common/cluster/discovery/k8s.py

```diff
@@ -202,6 +205,8 @@ async def initialize(self) -> None:
         self.update_node_data_cache_task = asyncio.create_task(
             self.update_node_data_cache()
         )
+        # Give some time to initialize and discover nodes
+        await asyncio.sleep(2)
```

Comment:
nit: adding an arbitrary time can often lead to issues when this runs in a slower environment. 
If possible, this should be replaced by an event-based waiter. e.g. is there a way for this code to check that the node discovery is over?

Reply:
@tarekziade good point indeed.

I've added a more "adaptative" way of checking it...
The thing is we don't know how many nodes are expected to be discovered.
However, IMO this is way better that not checking / waiting at all, which is what we had before.
WDYT?